rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -0,0 +1,131 @@
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        amount_raw = hex_or_bytes_to_int(tx_log.data[0:32])
+        interacted_address = hex_or_bytes_to_address(tx_log.topics[1])
+
+        for event in decoded_events:
+            amount = asset_normalized_value(amount_raw, event.asset)
+            if (
+                event.location_label == transaction.from_address == interacted_address is False or
+                (event.counterparty != ZERO_ADDRESS and event.balance.amount != amount)
+            ):
+                continue
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if event.counterparty == ZERO_ADDRESS:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    event.counterparty = CPT_CONVEX
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Spent {event.balance.amount} {event.asset.symbol} as part of withdrawal from {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Spent {event.balance.amount} {event.asset.symbol} as part of withdrawal from convex'  # noqa: E501
+                else:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.counterparty = CPT_CONVEX
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} into convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} into convex'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if tx_log.topics[0] in WITHDRAWAL_TOPICS:
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol} from convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol} from convex'  # noqa: E501
+                    event.counterparty = CPT_CONVEX
+                elif tx_log.topics[0] in REWARD_TOPICS:
+                    event.event_subtype = HistoryEventSubType.REWARD
+                    event.counterparty = CPT_CONVEX
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Receive reward {event.balance.amount} {event.asset.symbol} from convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
```

Comment:
Please rewrite to: "Claim AMOUNT TOKEN reward from convex POOL "

similarly below

Reply:
Done